File: PageObjectDotNet/AutomationPractice/PageObjects/AutomationPracticeStore/AuthenticationPage.cs
using AutomationPractice.PageObjects.AutomationPracticeStore.Objects;
using OpenQA.Selenium;
using OpenQA.Selenium.Support.PageObjects;

namespace AutomationPractice.PageObjects.AutomationPracticeStore
{
    public class AuthenticationPage : StorePage
    {
        private readonly By formErrors = By.CssSelector("div.form-error");


        public LoginForm AlreadyRegistered() 
        {
            var form = new LoginForm();
            PageFactory.InitElements(Driver, form);
            return form;
        }


        public NewAccount CreateAccount() 
        {
            return PageFactory.InitElements<NewAccount>(Driver);
        }

        public bool AreThereErrors() { return Driver.FindElements(formErrors).Count > 0; }

    }
}

File: PageObjectJava/src/test/java/PageObjects/AutomationPracticeStore/AuthenticationPage.java
package PageObjects.AutomationPracticeStore;

import Base.TestConfiguration;
import PageObjects.AutomationPracticeStore.Objects.LoginForm;
import PageObjects.AutomationPracticeStore.Objects.NewAccountEmail;
import org.openqa.selenium.By;
import org.openqa.selenium.support.PageFactory;

public class AuthenticationPage extends StorePage {

    private final By formErrors = By.cssSelector("div.form-error");

    public LoginForm alreadyRegistered() {
        return PageFactory.initElements(this._driver, LoginForm.class);
    }

    public NewAccountEmail createAnAccount() {
        return PageFactory.initElements(this._driver, NewAccountEmail.class);
    }

    public boolean areThereFormErrors(){
        return TestConfiguration.getDriver().findElements(formErrors).size() > 0;
    }

}
